addons/calendar/models/calendar.py

```diff
@@ -0,0 +1,1038 @@
+                final_date=fields.Datetime.from_string(data.get('start')) + timedelta(hours=values.get('duration', False) or data.get('duration'))
+            )
+
+            #do not copy the id
+            if data.get('id'):
+                del(data['id'])
+            return self.browse(real_event_id).copy(default=data)
+
+    @api.multi
+    def open_after_detach_event(self):
+        self.ensure_one()
+        new_id = self._detach_one_event()
+        return {
+            'type': 'ir.actions.act_window',
+            'res_model': 'calendar.event',
+            'view_mode': 'form',
+            'res_id': new_id.id,
+            'target': 'current',
+            'flags': {'form': {'action_buttons': True, 'options': {'mode': 'edit'}}}
+        }
+
+    @api.model
+    def _name_search(self, name='', args=None, operator='ilike', limit=100):
+        for arg in args:
+            if arg[0] == 'id':
+                for n, calendar_id in enumerate(arg[2]):
+                    if isinstance(calendar_id, basestring):
+                        arg[2][n] = calendar_id.split('-')[0]
+        return super(CalendarEvent, self)._name_search(name=name, args=args, operator=operator, limit=limit)
+
+    @api.multi
+    def write(self, values):
+
+        def _only_changes_to_apply_on_real_ids(field_names):
+            ''' return True if changes are only to be made on the real ids'''
+            for field in field_names:
+                if field in ['start', 'start_date', 'start_datetime', 'stop', 'stop_date', 'stop_datetime', 'active']:
+                    return True
+            return False
+        self._set_date(values)
+        ids = []
+
+        for one_ids in self.ids:
+            if isinstance(one_ids, (basestring, int, long)):
+                if len(str(one_ids).split('-')) == 1:
+                    ids = [int(one_ids)]
+                else:
+                    ids = [one_ids]
+        new_id = False
+
+        # Special write of complex IDS
+        for event_id in list(ids):
+            if len(str(event_id).split('-')) == 1:
+                continue
+
+            ids.remove(event_id)
+            real_event_id = calendar_id2real_id(event_id)
+
+            # if we are setting the recurrency flag to False or if we are only changing fields that
+            # should be only updated on the real ID and not on the virtual (like message_follower_ids):
+            # then set real ids to be updated.
+            if not values.get('recurrency', True) or not _only_changes_to_apply_on_real_ids(values.keys()):
+                ids.append(real_event_id)
+                continue
+            else:
+                event = self.browse(event_id)
+                if event.rrule:
+                    new_id = event._detach_one_event(values)
+
+        res = super(CalendarEvent, self.browse([int(event_id) for event_id in ids])).write(values)
+
+        # set end_date for calendar searching
+        record = self.browse(ids)
```

Comment:
record -> records and browse it like: ``` self.browse([int(event_id) for event_id in ids])) ``` as did at line 919 and change line 919 accordingly and use below wherever needed